### DISCUSSION TITLE
Open Source & Monetization

### DISCUSSION THREAD

**Discussion by zmlcc** [zed-industries/zed#6681] [Feature Requests]
Hi,
I've been following this project for a while and finally last week I could get a copy from [https://zed.dev/releases](https://zed.dev/releases).  Great work :+1:
However, I notice only the binary  can be downloaded but no source code. 
I want to know whether *Zed* will be completely open source in the furture? and which license will be used ? 
Or *Zed* would only be distributed as freeware ?

**Comment by l0go** [comment]
This is explained here: https://zed.dev/docs/faq
> **Will Zed be open source?**

> We'd like to be as open as possible while maintaining the ability to run a profitable business. This probably means some kind of "open core" model, where the editor is available under the GPL but other parts of our system remain proprietary. The timeline and exact strategy are still uncertain at this stage.

**Comment by iamnbutler** [comment]
As mentioned above, it's worth reading the faq (https://zed.dev/docs/faq), and let us know any additional questions you have. Additionally, Nathan covered this topic some more in his comments on this hacker news post as well: https://news.ycombinator.com/item?id=31668426

I'll likely close this issue once I have the chance to create a discussion topic around open source & monetization (discussions are a bit better for people to have threaded conversations.)

I think it is worth having an standing discussion around this topic, as lots of people will have thoughts and questions.

Thanks for the question!

**Reply by iamnbutler** [reply]
Some related pull quotes from @nathansobo from the above HN link:

> We plan to monetize collaboration. Zed is a collaborative platform disguised as a world-class code editor.

> Q. What's the reasoning behind this iteration being closed-source?
> This time we're on our own in the marketplace with no big revenue-generating company supporting us. We'd like to be as open as humanly possible while being able to build a defensible business around the editor.

**Comment by zmlcc** [comment]
@logo4poop  @iamnbutler  Thanks for the reminder. 
The balance between open source & monetization is really worth lots of discussion. Hope a clearer plan will be announced asap

**Comment by lukeed** [comment]
My 2 cents:

Release a base editor for OSS that's as lean & fast as it can be, but **without** any collaboration capabilities built into it. Zed is still extremely useful (and promising) without that, and the dev community needs something like it anyway. 

Then monetize the real-time collaboration, which means that the income can fund the maintenance & scaling for the infra needed for collaboration. 

This helps keep the base/OSS binary lean too, as there'd be no unused collab-bits inside the minimal core. Plus, it feels more premium for customers to be getting an exclusive download (that you may also verify/protect @ time of download).

**Reply by nuke-web3** [reply]
Yes, but I would lean further into 100% OSS that allowed people to operate collaborative features themselves (with what is likely to be non-negligible maintenance!). A reasonably close example of this is https://element.io/pricing - where people _may_ run smaller instances it is far and few between, and the company I work for now uses the higher tiers of hosting solutions and associated hosted services for a premium. 

The key aspect here is **on-prem** solutions where the isolation and ACLs to this mission critical data (chat for element/matrix, source code for zed collaboration servers) will be of very high value or even required to consider for use on many clients that are ready and willing to pay for it. Critically, with OSS collaboration servers, there is no zero-trust way to audit if there are no "back doors" or other privacy/security issues in operating or using them.

**Reply by stjepangolemac** [reply]
The essence of Zed is a "blazing fast, minimal, configurable / pluggable editor". Everything else is an extra and IMO asking too much from the team behind it.

This is like asking Zoom to give us both frontend and backend code and allow us to do the _hard_ task of running said code somewhere for 100% of functionality.

**Comment by maraisr** [comment]
Id second that — and a pretty cool game plan too. You capture the hearts and minds of all the developers out there, with a lean and fast IDE, without all the extra things. But with a few missing things to make it effective in business 🤩, and that is where you make your coin.

The GitHub model right; you make it fast/free/perfect for all the developers, and they sell it to their companies to use.

**Reply by l0go** [reply]
Well, GitHub isn't open source so I am unsure if that is a fair comparison. Maybe a model closer to Gitlab which has an open source core model.

**Reply by ijzerbroot** [reply]
For me personally, sublime text has shown that closed source is fine but you need a community if you are to have a living customisation ecosystem. ST's community looks quite stagnant by now but if it didn't then I'd most likely not even be looking for an alternative (yes, I even pay for it as for me it still beats VSCode which is just too busy and rickety-feeling, YMMV).
It would be totally impressive and laudable if you could manage to keep a base, extensible version of Zed free (and considering VSCodium's competition by now I guess it'll pretty much have to be..) and I am rooting for you to succeed. 
Whether collaboration-features by themselves would be enough to pay the bills I couldn't say but I suppose there's a lot more things that could potentially be plugged in that could be money-makers :)
Thank you already for publishing it for free at this stage!

**Comment by Aeolun** [comment]
Hmm, I'd be fine with paying a Jetbrains-esque monthly subscription, but I'm not sure if I'd pay it for the collaboration features alone if the editor was otherwise open. 

I'd pay it now, while still in beta, even though I already have access to the editor and would gain nothing extra, but the equation somehow changes when the editor becomes openly available.

**Reply by LeakyBucket** [reply]
An editor is a key tool for me, it is not something I want to rent it is something I feel it is important to "own".  Even if I happen to have an older version, access is far more important than being cutting edge current.

That said paying for the collaboration features etc.  Really sound like a sweet spot from a user perspective.  If I'm working with a group (dev team at a company etc) then the collaboration is a no brainer and also likely easier to justify opting into a subscription model.

**Reply by iamnbutler** [reply]
Exactly – Ideally you pay nothing, and your boss pays us 🙂 (This is just me talking conceptually, not making promises.)

**Reply by Aeolun** [reply]
I mean, when I pay my monthly subscription for Jetbrains I get all versions older than a year forever, which is more or less the same to me as buying a new 'forever' license every year (which was their previous model).

If it's in the same price range it's very easy to tell my boss to pay every year for a Zed license. They also have separate personal/corporate licenses, so while my boss would pay $800, I pay only $200/year, which I think is a reasonable amount to spend on a tool that I use literally 8 hours a day.

**Reply by Aeolun** [reply]
One thing that I really do not like is 'open-core'. It's either fully open, or it is not.

**Comment by sirinath** [comment]
I am not sure if GPL or copyleft is a good idea.

Best is to liberally / permissive open source and charge for hosting and infrastructure and setting up and managing on premise instances.

**Comment by keinsell** [comment]
About monetization, if features will be somewhat comparable to Webstorm (when it comes to project diagnosis, linters and refactorings) I'm open to support project by monthly subscription and would be really happy to drop JetBrains products.

**Comment by iamnbutler** [comment]
Not that this particular topic can be "answered", but just so people can see this that visit this topic:

We recently wrote a little bit about plans for open source. I imagine that Nathan will talk more about this in the future, but we are heads down building the tech that will enable us to do this now.

Here is the link to the blog post: https://zed.dev/blog/open-sourcing-zed-on-zed

**Reply by sirinath** [reply]
Can this be synched on GitHub too,

**Comment by KristijanZic** [comment]
> Not that this particular topic can be "answered", but just so people can see this that visit this topic:
> 
> We recently wrote a little bit about plans for open source. I imagine that Nathan will talk more about this in the future, but we are heads down building the tech that will enable us to do this now.
> 
> Here is the link to the blog post: https://zed.dev/blog/open-sourcing-zed-on-zed

I've red the blog post about open sourcing but it says nothing about open sourcing... Under what OSI approved license will it be released? Will there be a public git repository where all contributions are tracked and managed?

**Comment by Aeolun** [comment]
Since Zed is now mature enough for me to start using it as my main editor, is there also a way that I can start paying to use it? I'd very much dislike to start using it only for it to disappear due to lack of funding.

**Reply by Aeolun** [reply]
@KristijanZic I think after a while you realize that as great as open-source is, it doesn't come for free (looking at the graveyard of abandoned products I've used). I don't feel like there is a conflict between it being open-source and allowing me to pay for it's continued existence.

**Reply by KristijanZic** [reply]
@Aeolun here's what you said:
> Since Zed is now mature enough for me to start using it as my main editor, is there also a way that I can start paying to use it? I'd very much dislike to start using it only for it to disappear due to lack of funding.

You haven't mentioned open-source or any OSI license at all there. You asked if there was a way to pay for the product (implicitly as a closed source aka proprietary app). And hence I reacted with emojis that prompted you to write this.

>@KristijanZic I think after a while you realize that as great as open-source is, it doesn't come for free (looking at the graveyard of abandoned products I've used). I don't feel like there is a conflict between it being open-source and allowing me to pay for it's continued existence.

Firstly, it is not open source, as of today Zed remains a completely proprietary app and in case you haven't noticed this repo doesn't contain any Zed code. 
Secondly, I haven't said anywhere that there was a conflict between open-source and turning a profit. I'd have no problem with supporting it if it was actually open source under an OSI approved license, and not open-core but actually open-source.

**Comment by tml** [comment]
Doesn't the announcement on 2024.01.24 resolve this discussion? 

https://zed.dev/blog/zed-is-now-open-source

**Reply by KristijanZic** [reply]
Indeed it does 🥳

**Reply by SomeoneToIgnore** [reply]
Thank you for the reminder, I believe we can close it now and have new discussions open, if elaborations needed.
